torch/nn/modules/linear.py

```diff
@@ -120,10 +116,11 @@ def __init__(self, in1_features, in2_features, out_features, bias=True):
         self.reset_parameters()
 
     def reset_parameters(self):
-        bound = 1 / math.sqrt(self.weight.size(1))
+        std = 1 / math.sqrt(self.in1_features * self.in2_features)
```

Comment:
Why did you change the initialization for the bilinear, is there a reference for it?

Reply:
There's no reference for this, but it is the same math as for the regular linear layer. if you have two standard Gaussian signals, then this initialisation should make the output signal a standard Gaussian as well. 

The main point, however, is still that the desired standard deviation is used as a bound for the uniform, which results in "less desired" standard deviation of the generated samples!